libpod/runtime_pod_linux.go

```diff
@@ -95,7 +95,7 @@ func (r *Runtime) NewPod(ctx context.Context, options ...PodCreateOption) (*Pod,
 	if pod.config.UsePodCgroup {
 		logrus.Debugf("Got pod cgroup as %s", pod.state.CgroupPath)
 	}
-	if pod.HasInfraContainer() != pod.SharesNamespaces() {
+	if !pod.HasInfraContainer() && pod.SharesNamespaces() {
 		return nil, errors.Errorf("Pods must have an infra container to share namespaces")
```

Comment:
Can we retain the old behavior as a warning? "WARNING: Pod has infra container, but shares no namespaces" or similar?

Reply:
logrus debug?

Reply:
I would make it a Warnf